WalletWasabi.Fluent/Models/Wallets/WalletModel.cs

```diff
@@ -100,7 +108,56 @@ public WalletModel(Wallet wallet)
 
 	public IObservable<IChangeSet<IAddress, string>> Addresses { get; }
 
-	public string Name => Wallet.WalletName;
+	public string Name
+	{
+		get => GetName();
```

Comment:
`GetName()` performs disk IO which shouldn't be hidden behind a seemingly innocent, innocuous property getter.

This should read from disk only once and then cached.